src/jit/lclvars.cpp

```diff
@@ -1769,6 +1789,7 @@ void Compiler::lvaPromoteLongVars()
         {
             if (lvaGetPromotionType(varDsc->lvParentLcl) != PROMOTION_TYPE_INDEPENDENT)
             {
+                JITDUMP("Not promoting long local V%02u: field of dependently-promoted struct\n", lclNum);
```

Comment:
I don't know if these other no-promote cases happen, and if they do, what the implications are downstream. Yes, it seems like they probably guarantee SBCG bugs. We should probably assert that the type of decomposed lcl vars processed during decomp are 'int' and not 'long', e.g.